### DISCUSSION TITLE
Do we need to add "cleanupOutdatedCaches: true"

### DISCUSSION THREAD

**Discussion by muktihari** [openivity/openivity.github.io#122] [Q&A] [answered]
When we deploy an update, I think I need to refresh the page several time before the updated data is loaded properly. Does adding "cleanupOutdatedCaches: true" in "workbox" will help us achieve the immediate auto update without refreshing the page? Currently we don't specify the prop:

https://github.com/openivity/openivity.github.io/blob/e305f58ea387be3a47dd934c89726349bf162ba4/vite.config.ts#L29-L39

In this documentation, https://vite-pwa-org.netlify.app/guide/auto-update.html#cleanup-outdated-caches, says it's enabled by default, but in code doc it says otherwise: false. :cry: 

![image](https://github.com/user-attachments/assets/816aee3f-6b34-48b1-a0ae-913c3be9a921)

Any idea @raditzlawliet  ?

**Comment by raditzlawliet** [comment, marked as answer]
Hmm... I check out the web is have max-cache: 600. And every time we build, it always re-generate file with new name. That's why it should be fine for usual web.

But currently we use PWA to make cache and can be access when offline.
I re-read again, we need to add them unless if we're using `generateSW` strategy.
![image](https://github.com/user-attachments/assets/943d1757-23dc-49a7-972a-3dd1cfc97444)
![image](https://github.com/user-attachments/assets/cd1911dd-4f7d-4d75-8389-e5a7c92d1e45)

Try that options manually set `true`.
I also read something about background sync configuration, may can help https://vite-pwa-org.netlify.app/workbox/generate-sw

**Reply by muktihari** [reply]
I see, I'll send a PR to set `cleanupOutdatedCaches: true` first. Currently, I don't have time to change it into `generateSW` as I need to read more about it first and then still need to re-test it again, unless we really need it. But PR are welcome tho :sweat_smile:
